lxc/publish.go

```diff
@@ -227,6 +229,52 @@ func (c *cmdPublish) Run(cmd *cobra.Command, args []string) error {
 		req.ExpiresAt = expiresAt
 	}
 
+	existingAliases, err := GetCommonAliases(d, aliases)
+	if err != nil {
+		return fmt.Errorf(i18n.G("Error retrieving aliases: %w"), err)
+	}
+
+	if !c.flagReuse && len(existingAliases) > 0 {
+		names := []string{}
+		for _, alias := range existingAliases {
+			names = append(names, alias.Name)
+		}
+
+		return fmt.Errorf(i18n.G("Aliases already exists: %s"), strings.Join(names, ", "))
+	}
+
+	if c.flagReuse && len(existingAliases) > 0 {
+		visitedImages := make(map[string]interface{})
+		for _, alias := range existingAliases {
+			image, _, _ := d.GetImage(alias.Target)
+
+			// If the image has already been visited then continue
+			if image != nil {
+				_, found := visitedImages[image.Fingerprint]
+				if found {
+					continue
+				}
+
+				visitedImages[image.Fingerprint] = nil
+			}
+
+			// An image can have multiple aliases. An image should be removed
```

Comment:
Please can you explain this logic further, im not quite following the reasoning here. Thanks

Reply:
Thanks for the explanation, that makes sense now.

Reply:
So if an image being published reuses all the aliases from an existing image then that existing image is removed.

Reply:
Could you add a comment that to effect for future readers please?

Reply:
Done.